### PR TITLE
issue #30: documented naming convention

### LINKED ISSUE #30: Document naming conventions
To document our naming conventions for our resources.

### PR DESCRIPTION
**Pull request by k-allagbe**
#30
issue30-naming-conventions

added general convention and example for usual CLI accessible resources

Closes #30

### CODE REVIEW

File: Naming-Conventions/NAMING-RESOURCES.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- **Convention**: Use lowercase and separate words with dashes.
+- **Example**: `my-file.sh`
+
+### Folders/directories
+- **Convention**: Use lowercase and separate words with dashes.
+- **Example**: `my-folder`
+
+### Environment variables
+- **Convention**: Start with the uppercase repository name, followed by the actual variable name, separating words with underscores.
+- **Example**: `REPO_NAME_MY_ENV_VAR`
```
Comment (rngadam) [inline Naming-Conventions/NAMING-RESOURCES.md:22]: PROJECT_NAME and not REPO_NAME; we don't want 

MEMBRANE_BACKEND_ENVIRONMENT_VARIABLE 

...but:

MEMBRANE_ENVIRONMENT_VARIABLE.

we don't want to have to go back changing environment variable name if we moved from backend to frontend (in a fat client for example...)

File: Naming-Conventions/NAMING-RESOURCES.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+### Azure resource groups
+- **Convention**: Start with `rg-` followed by the name in lowercase, separating words with dashes.
+- **Example**: `rg-my-resource-group`
+
+### Azure resources
+- **Convention**: Use lowercase and separate words with dashes. Include the project name and type of resource.
+- **Example**: `my-project-storage-account`
```
Comment (rngadam) [inline Naming-Conventions/NAMING-RESOURCES.md:30]: project-storage-account

no doubt someone is going to interpret this as needing prefix my

maybe best would be to use bash environment variables so we have

PROJECT_NAME: your project name / codename such as membrane, nachet, louis, etc...

$(PROJECT_NAME)-storage-account

so for PROJECT_NAME=nachet and a storage account we have:

```
nachet-storage-account
```

File: Naming-Conventions/NAMING-RESOURCES.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (rngadam) [inline Naming-Conventions/NAMING-RESOURCES.md:31]: there are also a twist on SQL files where we require prefix 

`YYYY-mm-dd-description-of-script.sql`

see: https://github.com/ai-cfia/louis-db/tree/main/sql

**Review by rngadam — CHANGES_REQUESTED** [review]

File: Naming-Conventions/NAMING-RESOURCES.md (hunk 2 above)
Comment (rngadam) [reply Naming-Conventions/NAMING-RESOURCES.md:30]: expressed in shell-speak:

```
$ PROJECT_NAME=nachet; echo $PROJECT_NAME-storage-account
nachet-storage-account
```

File: Naming-Conventions/NAMING-RESOURCES.md (hunk 1 above)
Comment (k-allagbe) [reply Naming-Conventions/NAMING-RESOURCES.md]: done

File: Naming-Conventions/NAMING-RESOURCES.md (hunk 2 above)
Comment (k-allagbe) [reply Naming-Conventions/NAMING-RESOURCES.md]: done

File: Naming-Conventions/NAMING-RESOURCES.md (hunk 3 above)
Comment (k-allagbe) [reply Naming-Conventions/NAMING-RESOURCES.md:35]: done

**Review by rngadam — APPROVED** [review]

**Pull request merged**